frontend/src/app/main/ui/workspace/shapes/frame.cljs

```diff
@@ -6,125 +6,110 @@
-        old-shape (unchecked-get old-props "shape")
-
-        new-thumbnail? (unchecked-get new-props "thumbnail?")
-        old-thumbnail? (unchecked-get old-props "thumbnail?")
-
-        new-objects (unchecked-get new-props "objects")
-        old-objects (unchecked-get old-props "objects")
-
-        new-children (->> new-shape :shapes (mapv #(get new-objects %)))
-        old-children (->> old-shape :shapes (mapv #(get old-objects %)))]
-    (and (= new-shape old-shape)
-         (= new-thumbnail? old-thumbnail?)
-         (= new-children old-children))))
-
-(mf/defc frame-placeholder
-  {::mf/wrap-props false}
-  [props]
-  (let [{:keys [x y width height fill-color] :as shape} (obj/get props "shape")]
-    (if (some? (:thumbnail shape))
-      [:& frame/frame-thumbnail {:shape shape}]
-      [:rect.frame-thumbnail {:x x :y y :width width :height height :style {:fill (or fill-color cc/white)}}])))
-
-(defn custom-deferred
-  [component]
-  (mf/fnc deferred
-    {::mf/wrap-props false}
-    [props]
-    (let [shape (-> (obj/get props "shape")
-                    (select-keys [:x :y :width :height])
-                    (hooks/use-equal-memo))
-
-          tmp (mf/useState false)
-          ^boolean render? (aget tmp 0)
-          ^js set-render (aget tmp 1)
-          prev-shape-ref (mf/use-ref shape)]
-
-      (mf/use-effect
-       (mf/deps shape)
-       (fn []
-         (mf/set-ref-val! prev-shape-ref shape)
-         (set-render false)))
-
-      (mf/use-effect
-       (mf/deps render? shape)
-       (fn []
-         (when-not render?
-           (let [sem (ts/schedule-on-idle #(set-render true))]
-             #(rx/dispose! sem)))))
-
-      (if (and render? (= shape (mf/ref-val prev-shape-ref)))
-        (mf/jsx component props mf/undefined)
-        (mf/jsx frame-placeholder props mf/undefined)))))
-
-;; Draw the frame proper as a deferred component
-(defn deferred-frame-shape-factory
+(defn frame-shape-factory
   [shape-wrapper]
   (let [frame-shape (frame/frame-shape shape-wrapper)]
-    (mf/fnc defered-frame-wrapper
-      {::mf/wrap-props false
-       ::mf/wrap [#(mf/memo' % (mf/check-props ["shape" "childs"]))
-                  custom-deferred]}
-      [props]
-      (let [shape  (unchecked-get props "shape")
-            childs (unchecked-get props "childs")]
-        [:& frame-shape {:shape shape
-                         :childs childs}]))))
+    (mf/fnc frame-shape-inner
+      {::mf/wrap [#(mf/memo' % (mf/check-props ["shape" "fonts"]))]
+       ::mf/wrap-props false
+       ::mf/forward-ref true}
+      [props ref]
+
+      (let [shape         (unchecked-get props "shape")
+            fonts         (unchecked-get props "fonts")
+            childs-ref    (mf/use-memo (mf/deps (:id shape)) #(refs/children-objects (:id shape)))
+            childs        (mf/deref childs-ref)]
+
+        [:& (mf/provider embed/context) {:value true}
+         [:> shape-container #js {:shape shape :ref ref}
```

Comment:
If the props sturcture is known at compile time, it does not neet tu be prefixed with `#js`

Reply:
ok